File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    dic = 'AEIOU'
    wordList = []
    
    def dfs(cnt, temp):
        if cnt == 5:
            return
        for i in range(len(dic)):
            wordList.append(temp + dic[i])
            dfs(cnt + 1, temp+dic[i])
    
    dfs(0, '')
    
    return wordList.index(word)+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    static String[] words = {"A", "E", "I", "O", "U"};
    static List<String> list = new ArrayList<>();
    public int solution(String word) {
        int answer = 0;
        String str = "";

        dfs(str, 0);

        answer = list.indexOf(word);
        return answer;
    }

    public void dfs(String str, int len){
        list.add(str);
        if(len == 5)
            return;
        for(int i = 0; i < 5; i++){
            dfs(str + words[i], len+1);
        }
    }
}